### PR TITLE
Made Casus save and load automatically

### PR DESCRIPTION
**Pull request by SecondThread**
**Description**
This PR makes casus code save and load to the database. If you want to test it, you have to edit the tankID in saveCasus and loadCasus.

The reason this is rediculously annoying is that functions aren't saved in JSON. So to make sure we keep the functions attached to every Casus block in the tree, we need to 'revive' it (technical word, I didn't make it up). That is what I spent most of the time doing here and what the ungodly switch statement is for.

There really isn't a better way of doing that par than this unfortunately, since each of the different blocks needs to correspond to its own object, so that it has its own evaluate method.

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
Make a bulleted/numbered list of steps to verify this feature/bugfix. Include screenshots if necessary.

- Passes flow
- Saves and loads casus code. To verify, you need to
   - Create a new tank
   - Change the hardcoded id in saveCasus.js and loadCasus.js to the id of the tank you just created
   - then everything should work as you would expect

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

**Review by blaiadam — COMMENTED** [review]
Looks good, just want Emil or Baylor to review the route and controller changes.

File: src/backend/controllers/tankController.js [hunk 1]
```diff
@@ -7,82 +7,105 @@ const { validationResult } = require('express-validator');
...
+	await Tank.findById(req.params.tankId, function (err, tank) {
+		if (err){
+			console.error(err.message);
+			res.send(err);
+		}
+		// Update all fields of the tank except casus code
+		tank.tankName = req.body.tankName;
+		tank.userId = req.body.userId;
+		tank.components = req.body.components;
+		tank.isBot = req.body.isBot;
```
Comment (Derplime) [inline src/backend/controllers/tankController.js:82]: Is there a reason why we are setting `isBot` here? When the document is made, the field is automatically made `false` by default. Unless the reason here is that we want to utilize this API for setting bots as well as player tanks?

File: src/backend/controllers/tankController.js [hunk 2]
```diff
@@ -7,82 +7,105 @@ const { validationResult } = require('express-validator');
...
-            res.send(myUser.favoriteTankId);
-    });
+	await User.findById(req.user.id, 'favoriteTankId', (err: Error, myUser: User) => {
+		if(err) {
+			res.send(err);
+			console.log('could not find user');
+		} 
+		else
+			res.send(myUser.favoriteTankId);
+	});
```
Comment (Derplime) [inline src/backend/controllers/tankController.js:24]: I should honestly do callbacks like these for my past API implementations. Probably will do that when I do my stuff in the morning tomorrow.

**Review by Derplime — APPROVED** [review]
Just have that one question, other than that it looks good to me.

File: src/backend/controllers/tankController.js (hunk 1 above)
Comment (SecondThread) [reply src/backend/controllers/tankController.js:82]: I don't think we really need to be. This is how it was before, but I just left it how it was before

**Pull request merged**